dags/common/parsing/generic_parsing.py

```diff
@@ -19,3 +22,19 @@ def classification_numbers(arr, standard="PACS"):
 
 def free_keywords(arr, source="author"):
     return [{"source": source, "value": val} for val in arr if val]
+
+
+def fix_publication_date(pub_date):
+    if not pub_date:
+        return ""
+
+    def inner_fix(splitted):
+        return splitted if len(splitted) == 3 else inner_fix(splitted + ["1"])
+
+    splitted_date = inner_fix(pub_date.split("-"))
+    tmp_date = date(
+        int(splitted_date[0]),
+        int(splitted_date[1]),
+        int(splitted_date[2]),
+    )
+    return f"{tmp_date.year:04d}-{tmp_date.month:02d}-{tmp_date.day:02d}"
```

Comment:
Why not just `tmp_date.strftime('%Y-%m-%d')`?

Reply:
Because string formats are not the same in MacOS/Linux. Setting the date like this normalizes the format in all OS